### ISSUE TITLE
Training results are bad, collapse to noise like images, Why?

### ISSUE THREAD

**Issue by Oguzhanercan** [MFaceTech/InstantID#9]
Hi, I am trying to train the model with my own face recognition model, but results are really bad (My face recognation model outperforms from insgihtface's glintr360k, since it is trained with webface 42m).

![Screenshot from 2024-06-28 12-02-50](https://github.com/MFaceTech/InstantID/assets/64038202/2a78fd6f-adf2-4868-ad99-f4cd5773ae66)
![Screenshot from 2024-06-28 12-03-49](https://github.com/MFaceTech/InstantID/assets/64038202/064a0fee-3dce-4e4c-8c80-bd4da57da820)

Normally, at the training of controlnet, at the early stages, even it cannot control the output image with condition it is able to produce an image by prompt but in my case, the results are just make no sense.  What would be go wrong? I use FFHQ dataset captioned with blip, 70k samples. Training resolution is 128x128. Did not touch hyperparameters. Batch size is 48 with gradient accumulation. Used deepspeed bfloat16.

**Comment by chenxinhua** [comment]
First, I recommend not using your own face recognition network, but rather using the publicly available Insigface open-source network. The recognition accuracy has little impact on the generation effect. 


Second, I recommend using a training resolution of 1024x1024. Other resolutions will result in poor training results.



| |
陈新华
|
|
***@***.***
|


---- Replied Message ----
| From | Oğuzhan ***@***.***> |
| Date | 06/28/2024 14:14 |
| To | ***@***.***> |
| Cc | ***@***.***> |
| Subject | [MFaceTech/InstantID] Training results are bad, collapse to noise like images, Why? (Issue #9) |

Hi, I am trying to train the model with my own face recognition model, but results are really bad (My face recognation model outperforms from insgihtface's glintr360k, since it is trained with webface 42m).

Screenshot.from.2024-06-28.12-02-50.png (view on web)
Screenshot.from.2024-06-28.12-03-49.png (view on web)

Normally, at the training of controlnet, at the early stages, even it cannot control the output image with condition it is able to produce an image by prompt but in my case, the results are just make no sense. What would be go wrong? I use FFHQ dataset captioned with blip, 70k samples. Training resolution is 128x128. Did not touch hyperparameters. Batch size is 48 with gradient accumulation. Used deepspeed bfloat16.

**Comment by Oguzhanercan** [comment]
@chenxinhua  I tried 512x512, right now it generates good images, but identity space did not converged. It generates images belongs to same person, it is consistent in this manner, but the person is not the one I gave. The face recognation model I am using is arcface by insightface, but trained by me on a huge dataset (+50m images). I want to see the impact of face recognation models identity embeddings on generation quality, My model is outperforms when we compare it with insightface's open source glintr360k model.  

How many iteration did it take to converge in your experiments?

**Comment by jaebbb** [comment]
@Oguzhanercan Have you successfully completed the training?

**Comment by Oguzhanercan** [comment]
@jaebbb no, the training result aesthetically ok, but Identity preservation is not successful.

**Comment by jadechip** [comment]
@Oguzhanercan are you planning on open sourcing the weights of your trained arcface model? I was planning to use adaface but arcface would be easier.

**Comment by Oguzhanercan** [comment]
@jadechip unfortunately I cannot :/ But I can say that arcface outperforms adaface in my experiences.
